### DISCUSSION TITLE
Trovis 5573-0003

### DISCUSSION THREAD

**Discussion by FM-Audio** [Tom-Bom-badil/samson_trovis_557x#38] [General (de/en)]
Hallo,

Danke für dieses tolle Projekt!
Ich besitze eine 5573-0003 und bekomm sie einfach nicht zum laufen. Weder mit ioBroker noch mit Home Assistant.
Mein Nachbar hat die gleiche Anlage aber mit einem anderen Umsetzer. Er hat mir sogar sein HA Setup geschickt, welches ich fast 1:1 so übernehmen kann. Von dem DR302, habe ich auf der RX LED ein Dauerleuchten und auf dem TX ein Blinken. Ich gehe davon aus, dass beide blinken sollten? Danke für eure Hilfe, im Anhang findet ihr noch die Einstellungen aus dem Umsetzer. Unten werde ich auch noch ein Ausschnitt aus HA einfügen.

    Adapter
        Bezeichnung des verwendeten Adapters und Art Anbindung (PUSR DR302 DIN Schienen Modbus-Gateway)

    Trovis
        Trovis-Modell, wenn bekannt mit Unterkennung (Register 0, 5573-003)
        Firmware-Revision (2.25)
    

    Anbindung
        Buchse links seitlich (vorne hab ich keine gesehen)
        Wenn möglich: Kabelbelegung (Reihenfolge Trovis -> Adapterbeschriftung, z.B. 1 -> TxD, 8 -> RxD, 7 -> GND)
        Besonderheiten, sofern vorhanden (z.B. Adapter wurde auf Modbus-Client statt Modbus-Server eingestellt)?

![USR1](https://github.com/user-attachments/assets/fb22a094-8354-457a-9bf3-8d319176b436)
![USR2](https://github.com/user-attachments/assets/5c1dd9ca-68cd-4124-ba0d-754fa2b6693f)
![USR3](https://github.com/user-attachments/assets/7e4a151e-35ad-4861-85b0-141f7dc5a5ef)
![USR4](https://github.com/user-attachments/assets/c29358ce-eb47-4201-968a-a58f95774e06)

  name: modbus_hub
  type: tcp
  host: 192.168.178.141
  port: 8234
  
  delay: 0
  message_wait_milliseconds: 30
  timeout: 0.25

  sensors:
    - name: "Aussentemperatur"
      unique_id: Aussentemperatur
      device_address: 247
      address: 9
      scan_interval: 60
      input_type: holding
      data_type: int16
      scale: 0.1
      offset: 0
      device_class: temperature
      state_class: measurement
      unit_of_measurement: °C
    - name: "Heizung Vorlauf"
      unique_id: HeizungVorlauf
      device_address: 247
      address: 12
      scan_interval: 60
      input_type: holding
      data_type: int16
      scale: 0.1
      offset: 0

**Comment by Tom-Bom-badil** [comment]
Versuch mal Port 502 statt 8234.
/tom

**Reply by FM-Audio** [reply]
Leider ändert das nichts. Der Modbus Adapter in ioBroker zeigt auch connection true an. Das ist ja schonmal ein gutes Zeichen. Allerdings vermute ich, kommen da keine Werte rein. Hab mir die WIki 3mal hoch und runter durchgelesen und komm nicht drauf, wo der Fehler liegen könnte. 
![grafik](https://github.com/user-attachments/assets/7b7417fc-2d35-4ece-aac8-d01b4d7f693b)

**Reply by Tom-Bom-badil** [reply]
Nur zur Sicherheit: Auf dem Adapter bleibt 8234 eingestellt, in HA bzw ioBroker wird aber der ModbusTCP Standardport Port 502 verwendet. Korrekt?

Da Rx und Tx Werte anzeigen, Remote Rx und Tx aber nicht, vermute ich eine korrekte serielle Verkabelung Adapter <--> Trovis, aber in der Verbindung zwischen HA/ioBroker <--> Adapter hakt es irgendwo. Lässt sich 192.168.178.141 anpingen?

/tom

**Reply by FM-Audio** [reply]
Der USR hat als TCP Server die 192.168.178.141 mit dem Port 8234. Wenn ich einen anderen Port in ioBroker eintrage bekomme ich auch keine Verbindung. Als die müssen auf beiden Seiten gleich sein. Die 192.168.178.141 lässt sich anpingen. Auch die WebUI von dem USR ist aufrufbar. Vielleicht ist ein Fehler bei TCP oder RTU over TCP oder man muss Modbus TCP aktivieren. Aktuell hab ich nur Fragezeichen. Es ist zwar alles so toll dokumentiert, aber bei mir scheint eventuell etwas anders zu sein. Vielleicht gabs ja eine Revision?

**Reply by Tom-Bom-badil** [reply]
Bei mir läuft es exakt so (Screenshot gerade gemacht):

![image](https://github.com/user-attachments/assets/759c3782-8dbc-4a0a-a523-d012099f4084)

configuration.yaml
```
modbus:
  - name: trovis
    type: tcp
    host: 192.168.178.59
    port: 502
    sensors: !include_dir_merge_list trovis557x/sensors/
    binary_sensors: !include_dir_merge_list trovis557x/binary_sensors/
    switches: !include_dir_merge_list trovis557x/switches/
```

Siehe auch unter [HomeAssistant](https://github.com/Tom-Bom-badil/samson_trovis_557x/tree/master/HomeAssistant) im Code-Bereich.

/tom

Edit: Bild ausgetauscht - hatte die falsche Trovis erwischt. 😂

**Reply by FM-Audio** [reply]
Dankeschön, ich hab es mal auf 502 auf beiden Seiten abgeändert. Jetzt hab ich das Problem, dass der USR nicht mal mehr TX und RX zählt es kommt nix mehr rein. Das gibts doch nicht :D

**Comment by FM-Audio** [comment]
Ich gehe davon aus, dass die LED nicht so aussehen sollte: 
![grafik](https://github.com/user-attachments/assets/2bcf8049-390a-4dcd-baab-ae577178430c)

**Reply by Tom-Bom-badil** [reply]
Warte - der [DR302](https://www.pusr.com/products/din-rail-rs485-serial-to-ethernet-converter-usr-dr302.html?gad_source=1&gclid=Cj0KCQjwjNS3BhChARIsAOxBM6pd-XcWbZq4Z9mTwgxNOkkJ-Biwuvq8MMBvy9b_OPhPhjvQL7i8SxMaAhG3EALw_wcB) ist rein RS485 und kann kein RS232. Siehe die ersten 3 Absätze [hier](https://github.com/Tom-Bom-badil/samson_trovis_557x/wiki/Begriffe-und-'Basics'#schnittstelle-i-an-der-trovis---das-was-in-der-buchse-drin-ist). Die 5573 hat zwar beides, aber a) die RS485 Pins sind andere und b) wird da meines Wissens ein anderes Protokoll gesprochen.

/tom

**Reply by FM-Audio** [reply]
Oh, dann heißt das, den den DR302 kann ich gar nicht benutzen? Weil ich gesehen habe, dass der DR301 von jemand verwendet wurde.

**Reply by Tom-Bom-badil** [reply]
Genau. Der [DN301](https://www.pusr.com/products/din-rail-rs232-to-ethernet-converters-usr-dr301.html)=RS232, [DN302](https://www.pusr.com/products/din-rail-rs485-serial-to-ethernet-converter-usr-dr302.html)=RS485. Zwei unterschiedliche Modelle mit verschiedenen technischen ("elektrischen") Anschlüssen:

- RS232[TTL] = 3-Draht Direktanschluss voll-duplex (TxD **nur** Daten senden + RxD **nur** Daten empfangen + Masse), Spannung gemessen von TxD zu Masse oder von RxD zu Masse 0-5V mit Toleranz für 0-3.3V; 0V = logische "0", 5V = logische "1". Es können nur zwei Geräte direkt miteinander verbunden werden, wobei Senden/Empfangen gekreuzt sind (Senden geht an Empfangen der jeweils anderen Seite).
- RS485 = 2-Draht Bussystem halb-duplex (Spannung zwischen beiden Adern A und B -7...12 V ergibt logische 0 / 1; gleichzeitiges Senden **UND** Empfangen über diese 2 Leitungen). Anschluss von (theoretisch) bis zu 256 Geräten an 'A' und 'B' desselben Busses gleichzeitig. Das Protokoll (oft: Modbus) regelt, wer gerade 'spricht', und an wen gesendet wird. Für eine gute Übertragungsqualität inbesondere auf längeren Strecken wird oft noch Masse mit angeschlossen (oft als Schirm - meine KWL läuft zB seit 2014 problemlos auch ohne), daher kann man das mit einer 3-Draht 232 Verkabelung durchaus verwechseln.

![image](https://github.com/user-attachments/assets/15adc226-9a81-4e0d-80bc-5bf4f784d4fa)

Hoffe, das hilft für das Verständnis.
/tom

**Reply by FM-Audio** [reply]
Danke für die Aufklärung. Ich habe noch ein USR-W610 hier. Der kann RS232. Allerdings ist das so ein Anschluss wie bei einem VGA Monitor, da  brauch ich dann nochmal ein Adapter oder kann ich mir da was basteln? 
![grafik](https://github.com/user-attachments/assets/0c873b89-074a-4d7f-be56-d134163e5636)

**Reply by Tom-Bom-badil** [reply]
Der 9-Pin RS232-Stecker ist genormt ([mal Google fragen](https://www.google.com/search?q=rs232+9+pin)), da kann man auch schnell selbst was basteln (altes Kabel mit passendem Stecker vorausgesetzt).

/tom

**Reply by calle2010** [reply]
> Genau. Der [DN301](https://www.pusr.com/products/din-rail-rs232-to-ethernet-converters-usr-dr301.html)=RS232, [DN302](https://www.pusr.com/products/din-rail-rs485-serial-to-ethernet-converter-usr-dr302.html)=RS485. Zwei unterschiedliche Modelle mit verschiedenen technischen ("elektrischen") Anschlüssen:

Vielen Dank für die Erklärung. Ich möchte immer noch einen Elfin-Adapter anschließen und habe hier den EW11 liegen, der auch ein RS485-Adapter ist. Ich habe mir jetzt den EW10 bestellt, der RS232 kann.

Was ich mich noch Frage, was mit den Pegeln ist: RS232(TTL) arbeitet mit 0V und 5V, wenn ich es recht verstehe. RS232 normalerweise mit bis zu -15V/+15V. Ich frage mich jetzt, ob ich mit dem Elfin EW10 meinen Trovis grillen würde und einen Pegelwandler benötige?

**Reply by Tom-Bom-badil** [reply]
Vor kurzem hat hier jemand einen RS232 Adapter nur mit Pegelwandler zum Laufen gebracht: ([klick](https://github.com/Tom-Bom-badil/samson_trovis_557x/discussions/6#discussioncomment-10582502)).

Auf der [Webseite für den Elfin](http://www.hi-flying.com/elfin-ew10-elfin-ew11) finde ich aber folgendes:
![aa](https://github.com/user-attachments/assets/c5d1524c-431f-4490-bc28-ca89c3181929)

Nach meiner Lesart sieht das schon ziemlich nach TTL aus. Ich würd's erstmal so probieren ...

/tom

p.s. Hab meinen Elfin hier bestimmt schon seit 3 Jahren zu liegen und wollte den immer mal ausprobieren - es ist einfach nur zu wenig Zeit zum 'spielen'. :(

**Reply by calle2010** [reply]
Im User Manual (https://shop.marcomweb.it/images/virtuemart/product/Elfin-EW1X_User%20ManualV1.3(20200415).pdf?srsltid=AfmBOorVwTsTkJa4I3FHuNrUZEmfGdOVkWw8UpB7NWIyJ7iJsGe-ah3u)
2.3. RS232 Interface
Device RS232 does not support hardware flow control. The physical voltage is about ± 7V.
Andererseits müssten die das aus der Versorgungsspannung, die bei 5V beginnt, erzeugen. Ich habe halt keinen Bock, meine 5573 mit 14V zu zerschiessen, wenn es nicht sein muss.

**Reply by Tom-Bom-badil** [reply]
Schon klar. Brauchbaren und getesteten 232->TTL Adapter findest Du ja im Link oben, viel Erfolg! :)

/tom

_Edit: Was mich immer noch etwas verwirrt, ist, dass in oben verlinkter Anbindung tatsächlich ein step-down Adapter verwendet werden **musste**. Alle mir bekannten Trovis-Dokumente geben RS232 an - und nicht RS232-TTL. Bis jetzt bin ich immer davon ausgegangen, dass die Trovis an jeden Spec-gemässen Adapter angeschlossen werden kann, und nur nach 'unten' recht tolerant ist. Das scheint aber wohl nicht der Fall zu sein._

**Reply by calle2010** [reply]
Nachdem ich mein 40 Jahre altes Multimeter wieder zum Laufen bekommen habe, kann ich bestätigen, dass der Trovis 5573-0 RS232 mit TTL-Pegeln hat. Im Ruhezustand stehen TX und RX auf +5V, dass passt zu "The idle, no data state is high-voltage, or powered." https://en.wikipedia.org/wiki/Universal_asynchronous_receiver-transmitter Ich denke, an RX ist ein Pull-Up Widerstand.

Wenn ich den Elfin EW10 messe, ist der Pegel im Ruhezustand an TX -7V.

Es ist also nicht nur eine Anpassung der Spannung, sondern eine Umsetzung -7V / +7V  RS232 <=> +5V / 0V TTL

**Comment by FM-Audio** [comment]
Ich habe jetzt ein USR W610 dran hängen mit einem Adapter von RS232 zu RJ45. Dann gehe ich mit dem Netzwerkkabel von dem Adapter zu der Trovis. Der USR W610 hat noch etwas mehr Einstellungen, aber ich bekomm es immer noch nicht hin, dass Daten übertragen werden. In ioBroker steht die Verbindung auf true, aber die Holdings bleiben leer. Das Gleiche bei Home Assistant. Soll ich hier mal die Einstellungen der 610 posten? Vielleicht sieht man dann den Fehler.

**Reply by Tom-Bom-badil** [reply]
Klar, dazu ist die Diskussionsecke doch da. :)
Bitte auch mal Verkabelung (benutzte Pins) aufmalen - vielleicht fällt jemandem was auf ...

/tom

**Reply by FM-Audio** [reply]
Dankeschön =) vielleicht hab ich hier auch einen Denkfehler. Ich hab jetzt mit dem RS232 auf RJ45 Adapter einfach ein Netzwerkkabel mit allen PINS angeschlossen. Oder geht das garnicht? Muss vermutlich auch 1, 7, und 8 nehmen. Wenn ja wie macht man das am dümmsten, wenn auf beiden Seiten ein RJ45 Stecker dran ist?

**Reply by Tom-Bom-badil** [reply]
Der einfachste Weg: RJ45 Kabel nehmen, durchschneiden und wie [hier bei robhubi](https://github.com/Tom-Bom-badil/samson_trovis_557x/discussions/6#discussioncomment-4368286) beschalten (wer das nicht so oft macht, verwechselt beim RJ45 auch gern mal oben und unten, dann ist die Numerierung/Pin-Zählweise verkehrt;Masse ist dann an +5V und RxD TxD vertauscht). Auf der Adapterseite passenden Stecker besorgen und ebenfalls korrekt verkabeln.

Das Problem bei solchen von der Anleitung abweichenden Exoten-Adaptern ist leider, dass man ggf. experimentieren muss. Auch kann in diesem Fall ein Step-Down-Adapter auf TTL notwendig sein, siehe oben.

/tom

**Reply by FM-Audio** [reply]
Ich hab so einen Adapter hier gekauft. https://www.amazon.de/dp/B0BQJ26NPT?ref=ppx_yo2ov_dt_b_fed_asin_title Dann müsste ich ja einfach die Kabel trennen und 1,7 und 8 stehen lassen.

**Comment by FM-Audio** [comment]
Leute ich werd verrückt, die Heizung spuckt Werte aus. Danke für eure Hilfe! 
![image](https://github.com/user-attachments/assets/12285061-cc48-40c4-a45f-d61f7a788ab1)

**Reply by Tom-Bom-badil** [reply]
Sehr schön, Glückwunsch! :)
Wird der Step-Down-Adapter auf TTL denn nun wirklich benötigt, oder geht es auch ohne?

/tom

**Reply by FM-Audio** [reply]
Also ich denke ich brauche ihn. Bei der Anlage scheint eh alles etwas anders zu sein. Ich hab zb. 40000er Register. Und meine Vor und Rücklauftemperatur ist gefühlt falsch herum. 
![grafik](https://github.com/user-attachments/assets/bfb4ee02-0b87-42a5-9502-d7eb5c46b731)

**Reply by Tom-Bom-badil** [reply]
Die Art, wie Modbus-Register adressiert werden (Start bei 0 oder 40.001), ist u.a. von der verwendeten Modbus-Bibliothek abhängig. Ich vermute, ioBroker verwendet da was anderes als z.B. HA (dort basiert die Implementierung auf [pymodbus](https://pymodbus.readthedocs.io/en/latest/source/readme.html), welches ich damals auch für mein SmartHomeNG Plugin genommen habe). Ich würde fast wetten, dass in HA auch bei Dir die Register mit 0 starten.

Zu den Temperaturen: Passt schon. Vorlauf ist der (eher kühlere) Sekundärkreis, Rücklauf ist der (eher wärmere) Primärkreis. Wahrscheinlich war kurz vorher auch noch WW-Erwärmung. Schau Dir mal die Bilder an, die wir grad in dem anderen Thread veröffentlich haben, und wo da die Sensoren sitzen, dann wird es klarer. Ist das eine Kompaktstation?

/tom

**Reply by FM-Audio** [reply]
Ja du hast Recht, bei ioBroker geht es mit 40.001 los und bei HA bei 0. Es ist eine Fernwärmestation, also an sich nicht viel Technik drin. Aber ab und zu verschluckt er sich mit den Werten. Bei meinem Nachbar soll das übrigens auch so sein. Ist zwar nicht schlimm, aber da hast in den Langzeitverläufen immer mal wieder so Ausreizer drin. Hatte jetzt mal 500Grad laut Anzeige, da ist das Duschen ausgefallen, war mir doch ein bisschen zu warm. :D

**Reply by Tom-Bom-badil** [reply]
Bekanntes Phänomen - Rücklauf aus der 'Duschhölle' beträgt 2.723 °C. ;) Von den 21 Pewo-Stationen bei uns im Wohngebiet (+1 große) hat den Effekt ca die Hälfte (un-)regelmäßig.

Hat aber nach meiner Erfahrung nichts mit dem Regler selbst zu tun. Schuld sind nach meiner Einschätzung minderwertige Sensoren bzw Kontaktprobleme / sporadische Wackelkontakte (zB durch Oxydation). Meist schafft ein Wackeln an den Anlegefühlern für eine Weile Abhilfe.

Hintergrund: Viele Fühler sind nur temperaturabhängige Widerstände (siehe zB [hier](https://www.temperaturmesstechnik.de/fileadmin/user_upload/pdf/tmh_pt1000_tabelle.pdf) die Tabelle für den meist benutzten PT1000). Der Regler misst nur den Widerstandswert des Fühlers. Ein Kontaktproblem dort = plötzlich sehr hoher Widerstand = für den Regler eine sehr hohe Temperatur. Ganz simpel.

/tom

_p.s. / edit: Ärgerlich wird das ganze nur, wenn es Dir die Plots 'zerschießt', weil Ymax plötzlich 2.000°C sind, im Normalbetrieb aber Ymax bei vielleicht 70° liegen würde (=kein Graph mehr zu sehen). Noch ärgerlicher wird es aber, wenn der Regler längere Zeit denkt, Dein Rücklauf zum Versorger beträgt 2.000 °C, der aber auf zB 50° begrenzt ist - dann macht der nämlich das Stellventil zu, und Du stehst ohne Heizung und WW da._
